goalie/mesh_seq.py

```diff
@@ -191,35 +232,79 @@ def plot(
             axes = axes[0]
         return fig, axes
 
-    def get_function_spaces(self, mesh: firedrake.MeshGeometry) -> Callable:
+    def get_function_spaces(self, mesh):
+        """
+        Construct the function spaces corresponding to each field, for a given mesh.
+
+        :arg mesh: the mesh to base the function spaces on
+        :type mesh: :class:`firedrake.mesh.MeshGeometry`
+        :returns: a dictionary whose keys are field names and whose values are the
+            corresponding function spaces
+        :rtype: :class:`dict` with :class:`str` keys and
+            :class:`firedrake.functionspaceimpl.FunctionSpace` values
+        """
         if self._get_function_spaces is None:
             raise NotImplementedError("'get_function_spaces' needs implementing.")
         return self._get_function_spaces(mesh)
 
-    def get_initial_condition(self) -> dict:
+    def get_initial_condition(self):
+        r"""
+        Get the initial conditions applied on the first mesh in the sequence.
+
+        :returns: the dictionary, whose keys are field names and whose values are the
+            corresponding initial conditions applied
+        :rtype: :class:`dict` with :class:`str` keys and
+            :class:`firedrake.function.Function` values
+        """
         if self._get_initial_condition is not None:
             return self._get_initial_condition(self)
         return {
             field: firedrake.Function(fs[0])
             for field, fs in self.function_spaces.items()
         }
 
-    def get_form(self) -> Callable:
+    def get_form(self):
+        """
+        Get the function mapping a subinterval index and a solution dictionary to a
+        dictionary containing parts of the PDE weak form corresponding to each solution
+        component.
+
+        :returns: the function for obtaining the form
```

Comment:
This might apply to all the `get_form`, `get_solver`, ... methods which return a function for obtaining their respective components. If not found elsewhere, it would be good to include the expected return type more explicitly as you do elsewhere with `:rtype:`  to give the user more guidance when constructing the nested function.

Reply:
How about 1c74898?